codec/src/main/java/io/netty/handler/codec/protobuf/util/ProtobufVarint32FrameDecoderBase.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2012 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.protobuf.util;
+
+import com.google.protobuf.CodedInputStream;
+
+import java.io.IOException;
+import java.util.List;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.ByteToMessageDecoder;
+import io.netty.handler.codec.CorruptedFrameException;
+
+/**
+ * A decoder that splits the received {@link ByteBuf}s dynamically by the
+ * value of the Google Protocol Buffers
+ * <a href="http://code.google.com/apis/protocolbuffers/docs/encoding.html#varints">Base
+ * 128 Varints</a> integer length field in the message.  For example:
+ * <pre>
+ * BEFORE DECODE (302 bytes)       AFTER DECODE (300 bytes)
+ * +--------+---------------+      +---------------+
+ * | Length | Protobuf Data |----->| Protobuf Data |
+ * | 0xAC02 |  (300 bytes)  |      |  (300 bytes)  |
+ * +--------+---------------+      +---------------+
+ * </pre>
+ *
+ * @see CodedInputStream or @see CodedInputByteBufferNano
+ */
+public abstract class ProtobufVarint32FrameDecoderBase extends ByteToMessageDecoder {
+
+    // TODO maxFrameLength + safe skip + fail-fast option
```

Comment:
It's carried over from previous implementation and the main reason i created the abstraction for this class. I might address the TODO but I think it's better left for an separate PR.